### PR TITLE
Dont count items that are previously clicked in hitrate calc as default

### PR DESCRIPTION
**Pull request by simeneide**
Tested on a couple of cases I had handy and the hitrate is lowered by a reasonable amount.

Breaking change: Changes the previous default behaviour to not including previous clicks, but left as an option in the hitrate module.

### CODE REVIEW

**Comment by NegatioN** [comment]
Couldn't you just count hits like normal, and then subtract the sum of "noCount hits" at the end? That would make it a bit simpler, and make the need to worry about masking noCounts and stuff fall away

**Comment by simeneide** [comment]
Unfortunately, no. What we really want to do is to exclude the items clicked in training data from being allowed to be recommended. For example, if a user clicked on item 5 in training and in test period, we cannot first count all matches (2) and then subtract the training match (1). That would leave us with 1 match, but the desired result is 0 matches.

File: recsys_slates_dataset/lightning_helper.py [hunk 1]
```diff
@@ -128,17 +128,28 @@ class Hitrate(pl.Callback):
...
         batch_recs = pl_module.recommend_batch(batch,num_rec= self.num_rec,t_rec=-1).detach().cpu()
+
+        # If a recommendation already appears in the training click sequence, remove it from recommendations.
+        # It is removed by setting the recommendation to a negative number ( :rolling_eyes:, i know),
+        # which will not be counted. This makes it faster&paralleizeable in the np.intersect1d part.
+        if self.remove_already_clicked:
+            dont_count_clicks = (batch['click']*(~batch['phase_mask'])).detach().cpu()
+            for n in range(batch_recs.size(1)):
+                rec_clicked_item = (batch_recs[:,n].unsqueeze(1)==dont_count_clicks).max(dim=1)[0]
+                batch_recs[rec_clicked_item,n] = -(1+n)
```
Comment (NegatioN) [inline recsys_slates_dataset/lightning_helper.py]: This is the masking value, right? Is there a reason you're using `-(1+n)` and not just using `-1` here?
I don't think using a negative number as a mask is that bad though, by the way :smile:

Comment (simeneide) [reply recsys_slates_dataset/lightning_helper.py]: correct. 
Uhm, my thinking yesterday was that it would match with other `-1` values elsewhere, but I guess that will never be the case as `batch['click']` never will have any negative values in it anyways... I think my misthinking here must have been due to a copy error during test where I modified the click data.

Comment (NegatioN) [reply recsys_slates_dataset/lightning_helper.py]: You could just pick some unlikely number like -1024 or something i suppose?
I just figured it's a bit more readable if the value is the same in every row of the data. :)

**Review by NegatioN — APPROVED** [review]
LGTM after mask-change
